fix: upgrade `textextensions` version so links to `XAML` files in preview open in editor

SeriousBug opened the pull request:
- DEPENDENCY_UPDATE: @dendronhq/plugin-core updated textextensions from 5.14.0 to 5.15.0

This change includes my contribution adding `XAML` as a text extension to the list, which will allow the Dendron preview to open XAML files when a linked to one is clicked in the preview.

The upgrade command also pulled in minor version upgrades to `nextjs`, but I left them in since they should be harmless.

#1695

kevinslin commented:
actually had to downgrade nextjs because the swc module we're causing issues in github pipelines. could we revert the nextjs changes and add "12.1.0" in nextjs-template `package.json` without the `^`?

SeriousBug commented:
Got it, sure! I fixed the versions to what they were in the lockfile, there's no change in the lockfile now. Git still shows changes, but you can see it's because I changed `^11.1.2` to `11.1.4` since that's what it already was in the lockfile.
